packages/processors/src/strategy/donationVotingMerkleDistributionDirectTransfer/handlers/allocated.handler.ts

```diff
@@ -0,0 +1,198 @@
+import { Address, encodePacked, getAddress, keccak256 } from "viem";
+
+import { Application, Changeset, Donation, Round } from "@grants-stack-indexer/repository";
+import { ChainId, getToken, ProcessorEvent, Token } from "@grants-stack-indexer/shared";
+
+import { getTokenAmountInUsd, getUsdInTokenAmount } from "../../../helpers/index.js";
+import {
+    ApplicationNotFound,
+    IEventHandler,
+    ProcessorDependencies,
+    RoundNotFound,
+    UnknownToken,
+} from "../../../internal.js";
+import { ApplicationMetadata, ApplicationMetadataSchema } from "../../../schemas/index.js";
+
+type Dependencies = Pick<
+    ProcessorDependencies,
+    "roundRepository" | "applicationRepository" | "pricingProvider"
+>;
+
+export class OriginMissing extends Error {
+    constructor() {
+        super("Origin field is required");
+    }
+}
+
+export class MetadataParsingFailed extends Error {
+    constructor(additionalInfo?: string) {
```

Comment:
i thought maybe they're only specific to this handler,  but if later i need it for another Handler, i can refactor and move to exceptions, wdyt?

Reply:
i agree with jahabeebs, lets keep them on its specific directory to avoid noise on this file... we can have `exceptions/handlers/DVMD....` or some similar directory structure, wdyt ?